plugins/inputs/zipkin/handler.go

```diff
@@ -106,30 +118,25 @@ func (s *SpanHandler) Spans(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
-func unmarshalThrift(body []byte) ([]*zipkincore.Span, error) {
-	buffer := thrift.NewTMemoryBuffer()
-	if _, err := buffer.Write(body); err != nil {
-		return nil, err
+// ContentType returns a Decoder that is able to produce Traces from bytes.
+// Failure should yield an HTTP 415 (`http.StatusUnsupportedMediaType`)
+// If a Content-Type is not set, zipkin assumes application/json
+func ContentType(r *http.Request) (codec.Decoder, error) {
```

Comment:
I would consider renaming this function, the current name does not indicate what the function does.